org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLCompletions.java

```diff
@@ -691,74 +704,83 @@ private void collectAttributeValueSuggestions(int valueStart, int valueEnd, Comp
+			} else {
+				if(isContent) {
+					editRange = document.getTrimmedRange(node.getStart(), node.getEnd());
+				} 
+				if(editRange == null) {
+					editRange = getReplaceRange(node.getStart(), node.getEnd(), request);
+				}
+			}
 		} catch (BadLocationException e) {
-			LOGGER.log(Level.SEVERE, "While performing getReplaceRange for DTD !ELEMENT completion.", e);
+			LOGGER.log(Level.SEVERE, "While performing getReplaceRange for DTD completion.", e);
 		}
+				
+		elementDecl.setTextEdit(new TextEdit(editRange, "<!ELEMENT ${1:element-name} (${0:#PCDATA})>"));
+		elementDecl.setDocumentation("<!ELEMENT element-name (#PCDATA)>");
+		
 		response.addCompletionItem(elementDecl);
 
 		// Insert DTD AttrList Declaration
 		// see https://www.w3.org/TR/REC-xml/#attdecls
 		CompletionItem attrListDecl = new CompletionItem();
 		attrListDecl.setLabel("Insert DTD Attributes list declaration");
 		attrListDecl.setKind(CompletionItemKind.EnumMember);
-		attrListDecl.setFilterText("<!ATTLIST");
-		attrListDecl.setInsertTextFormat(InsertTextFormat.Snippet);
+		attrListDecl.setFilterText("<!ATTLIST ");
+		attrListDecl.setInsertTextFormat(insertFormat);
 		startOffset = request.getOffset();
-		try {
-			Range editRange = getReplaceRange(startOffset, startOffset, request);
-			attrListDecl.setTextEdit(new TextEdit(editRange,
-					"<!ATTLIST ${0:element-name} ${1:attribute-name} ${2:ID} ${3:#REQUIRED} >"));
-			attrListDecl.setDocumentation("<!ATTLIST element-name attribute-name ID #REQUIRED >");
-		} catch (BadLocationException e) {
-			LOGGER.log(Level.SEVERE, "While performing getReplaceRange for DTD !ATTLIST completion.", e);
-		}
+		
+		attrListDecl.setTextEdit(new TextEdit(editRange,
+				"<!ATTLIST ${1:element-name} ${2:attribute-name} ${3:ID} ${0:#REQUIRED}>"));
+		attrListDecl.setDocumentation("<!ATTLIST element-name attribute-name ID #REQUIRED>");
+		
 		response.addCompletionItem(attrListDecl);
 
 		// Insert Internal DTD Entity Declaration
 		// see https://www.w3.org/TR/REC-xml/#dt-entdecl
 		CompletionItem internalEntity = new CompletionItem();
 		internalEntity.setLabel("Insert Internal DTD Entity declaration");
 		internalEntity.setKind(CompletionItemKind.EnumMember);
-		internalEntity.setFilterText("<!ENTITY");
-		internalEntity.setInsertTextFormat(InsertTextFormat.Snippet);
+		internalEntity.setFilterText("<!ENTITY ");
+		internalEntity.setInsertTextFormat(insertFormat);
 		startOffset = request.getOffset();
-		try {
-			Range editRange = getReplaceRange(startOffset, startOffset, request);
-			internalEntity.setTextEdit(new TextEdit(editRange, "<!ENTITY ${0:entity-name} \"${1:entity-value}\" >"));
-			internalEntity.setDocumentation("<!ENTITY entity-name \"entity-value\" >");
-		} catch (BadLocationException e) {
-			LOGGER.log(Level.SEVERE, "While performing getReplaceRange for DTD !ENTITY completion.", e);
-		}
+		
+		internalEntity.setTextEdit(new TextEdit(editRange, "<!ENTITY ${1:entity-name} \"${0:entity-value}\">"));

```

Comment:
you can't use placeholders if snippets are not supported. You need to check insertFormat.